### DISCUSSION TITLE
How to keep data up to date without rebuilding whole project?

### DISCUSSION THREAD

**Discussion by mroddev** [nuxt/nuxt#21871] [Questions]
Hi.

I have a blog type website built with Nuxt 3 and using Wordpress as a headless CMS and getting the backoffice data with nuxt-apollo via graphql.
As I have infinite scrolling I'm currently prerending all routes using the following inside defineNuxtConfig:
```
hooks: {
    async "nitro:config"(nitroConfig) {
      if (nitroConfig.dev) {
        return;
      }
      let slugs = await dynamicRoutes();
      if (nitroConfig.prerender?.routes) {
        nitroConfig.prerender.routes.push(...slugs);
        return;
      }
    },
  },
```

This works fine but I'm starting to get bigger and bigger deploy times on Vercel and even worse, every time I create/update or delete an entry in Wordpress I have to rebuild the whole project for those changes to be applied in the frontend.

I've tried enabling **ISR**  and removing the prerendering shown above but it doesn't work and keeps showing the server cached page. The only way to see updated content without rebuilding is by doing a client side navigation from a nuxt-link.

This was my nuxt.config with ISR enabled:
```
export default defineNuxtConfig({
  ...
  routeRules: {
      "/**": { isr: true },
  }
  ...
})
```

**Comment by warflash** [comment]
`isr: true` will render the page once on request and then cache it until you redeploy your app.
if you want the cache to be revalidated in the background you want to use e.g. `isr: 60`

**Comment by mroddev** [comment]
> `isr: true` will render the page once on request and then cache it until you redeploy your app. if you want the cache to be revalidated in the background you want to use e.g. `isr: 60`

Still no luck, tried with ```isr: 20``` and data is not updated on refresh. I am starting starting to think this might be related to nuxt-apollo rather than Nuxt/Nuxt rendering itself...

**Comment by DidoMarchet** [comment]
Hi, did you find a solution?
I've got my application Nuxt3 and Apollo deployed in Netlify with the netlify-builder.
I'm not able to refresh the content when something change into the cms. The only way is rebuild the app.
I try to set some limit to the cchew but nothing happens.

```
  routeRules: {
    '/typology/**': { swr: 120 },
    'product-family/**': { swr: 120 },
    '/product/**': { swr: 120 },
  },
```

Kind regards,
Davide

**Reply by mroddev** [reply]
Hi. 
No I haven't found a solution yet.I browsed a bit and there seems to be some issues with the nuxt-apollo cache in the current version but still, not entirely sure if the problem is even there in the first place. 

I was trying to change my project to [nuxt-graphql-client](https://github.com/Diizzayy/nuxt-graphql-client) to see if the problem still persisted but haven't tried it yet.

**Reply by DidoMarchet** [reply]
Hi,

I'm not sure which CI/CD system you're using, but with Netlify, if you set 'netlify-edge' as your preset, it should work as expected. So, I don't believe the issue is related to Apollo.

Here's my configuration:
```
nitro: {
  routeRules: {
    '/**': { isr: true }
  },
  // https://nitro.unjs.io/deploy/providers/netlify
  preset: 'netlify-edge',
}
```

One thing to note if you're using 'netlify-edge' is that you need to patch the graphql library: https://github.com/nuxt-modules/apollo/issues/525. This needs to be done for both .js and .mjs files.

Regrads,
Davide

**Comment by mroddev** [comment]
> Hi,
> 
> I'm not sure which CI/CD system you're using, but with Netlify, if you set 'netlify-edge' as your preset, it should work as expected. So, I don't believe the issue is related to Apollo.
> 
> Here's my configuration:
> 
> ```
> nitro: {
>   routeRules: {
>     '/**': { isr: true }
>   },
>   // https://nitro.unjs.io/deploy/providers/netlify
>   preset: 'netlify-edge',
> }
> ```
> 
> One thing to note if you're using 'netlify-edge' is that you need to patch the graphql library: [nuxt-modules/apollo#525](https://github.com/nuxt-modules/apollo/issues/525). This needs to be done for both .js and .mjs files.
> 
> Regrads, Davide

Thanks for the feedback!

I haven't tried with 'netlify-edge' or in my case 'vercel-edge' since I'm deploying to Vercel. When I tried it exceeded Vercel edge function limit of 1MB😕

I have looked at some examples such as @danielroe [nuxt-vercel-isr](https://github.com/danielroe/nuxt-vercel-isr) and he doesn't seem to be using the nitro preset, atleast I can't really find any information about it.

Is it really mandatory to enable nitro's preset for ISR to work in vercel/netlify/others? I'm sorry if this is obvious but I'm a bit confused about all of this.

**Reply by logotip4ik** [reply]
I actually not sure if you need to provide preset at all. Nitro auto detects environment ([at least tries to](https://nitro.unjs.io/deploy#zero-config-providers)) and will output correct build. But by default it will not use `edge` preset, and in my case - vercel with normal serverless output does work with `isr`. It also could be that something overrides `Cache-Control` header ([vercel respects these headers](https://vercel.com/docs/concepts/edge-network/caching#cache-control-options)). That should be same as your `isr` or not present at all
